torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -43,6 +43,8 @@ RRefForkData::RRefForkData(
       type_str_(std::move(type_str)) {}
 
 py::tuple RRefForkData::toPyTuple() const {
+  // add GIL as it is contructing a py::object
+  pybind11::gil_scoped_acquire ag;
```

Comment:
Do we need GIL for `RRefForkData::fromPyTuple` as well? or it has already acquired GIL, how?

Reply:
sounds good, let me add GIL for fromPyTuple as well

Reply:
it's a const reference of the `py::tuple` and does not seem to create/delete `py::object`. Is it still necessary to use GIL in `fromPyTuple`.

Reply:
I guess GIL here can help avoid underneath python object pointed by pyTuple could be mutated?

Reply:
@mrshenli Accessing (both read and write) a Python object also requires GIL, otherwise, other thread could get GIL and invalidate that Python object.